junebug/workers.py

```diff
@@ -242,8 +262,9 @@ def request_failed(resp):
     return resp.code < 200 or resp.code >= 300
 
 
-def post(url, data):
+def post(url, data, timeout):
     return treq.post(
         url.encode('utf-8'),
         data=json.dumps(data, cls=JSONMessageEncoder),
-        headers={'Content-Type': 'application/json'})
+        headers={'Content-Type': 'application/json'},
+        timeout=timeout)
```

Comment:
I think in `post()` is a good place; we need to do the same thing for both messages and events (catch and log the error), so we would just be repeating ourselves.